burnman/planet.py

```diff
@@ -0,0 +1,216 @@
+            layer.thickness = layer.outer_radius - layer.inner_radius
+            current_radius = layer.outer_radius
+            slices[0] += 1 # make the boundaries one meter thick
+            layer.n_start = n
+            layer.n_end = n + layer.n_slices
+            n += layer.n_slices
+            self.radial_slices = np.append(self.radial_slices, slices)
+
+        self.radial_slices[0] = 0.0  # overwrite inner-most radius to zero
+        self.radius = current_radius
+
+        # initial pressure guess: linear as a function of radius
+        max_p = 350e9
+        self.pressures = np.array([max_p * (1.0 - r / self.radius) for r in self.radial_slices])
+        self.temperatures = [self.get_layer_by_radius(r).temperature(r) for r in self.radial_slices]
+        self.densities = np.empty_like(self.pressures)
+        self.gravity = np.empty_like(self.pressures)
+        self.bulk_sound_speed = np.empty_like(self.pressures)
+        self.shear_velocity = np.empty_like(self.pressures)
+
+        self.averaging_scheme = averaging_schemes.VoigtReussHill()
+
+        last_center_pressure = 0.0
+        for i in range(n_max_iterations):
+            if verbose:
+                print("on iteration %d" % (i+1))
+            self._evaluate_eos()
+
+            self._compute_gravity(self.densities, self.radial_slices)
+            self._compute_pressure(self.densities, self.gravity, self.radial_slices)
+
+            # compute relative error (pressure in the center of our planet)
+            rel_err = abs(last_center_pressure - self.pressures[0]) / self.pressures[0]
+            if verbose:
+                print("  relative central core pressure error between iterations: %e" % rel_err)
+
+            if rel_err < 1e-5:
+                break
+            last_center_pressure = self.pressures[0]
+
+        self.mass = self._compute_mass()
+
+        self.moment_of_inertia = self._compute_moment_of_inertia()
+        self.moment_of_inertia_factor = self.moment_of_inertia / self.mass / self.radial_slices[-1] / self.radial_slices[-1]
+
+    def get_layer(self, name):
+        for layer in self.layers:
+            if layer.name == name:
+                return layer
+        raise LookupError()
+
+    def get_layer_by_radius(self, radius):
+        for layer in self.layers:
+            if layer.outer_radius >= radius:
+                return layer
+        raise LookupError()
+
+    def _evaluate_eos(self):
+        # evaluate each layer separately
+        for layer in self.layers:
+            mypressures = self.pressures[layer.n_start: layer.n_end]
+            mytemperatures = self.temperatures[layer.n_start: layer.n_end]
+
+            density, vs, vphi = layer.rock.evaluate(['density', 'v_s', 'v_phi'], mypressures, mytemperatures)
+
+            self.densities[layer.n_start: layer.n_end] = density
+            self.bulk_sound_speed[layer.n_start: layer.n_end] = vs
+            self.shear_velocity[layer.n_start: layer.n_end] = vphi
+
```

Comment:
add a function to evaluate any properties
def evaluate(self, variables, pressures= none, depths=none):

If no pressures or depths are defined it returns along the internal depth list.